build_posix/aclocal/strict.m4

```diff
@@ -90,6 +90,12 @@ AC_DEFUN([AM_CLANG_WARNINGS], [
 	w="$w -Wno-reserved-id-macro"
 	w="$w -Wno-zero-length-array"
 
+	# Added when switching to building with clang 3.4.2
+	w="$w -Wno-parentheses-equality"
+	w="$w -Wno-self-assign"
+	w="$w -Wno-unreachable-code"
+	w="$w -Wno-unreachable-code-return"
```

Comment:
This change is sufficient for margay and kodkod.  To get the build working with the version of clang on tinderbox, we'd also need:
```
        w="$w -Wno-array-bounds"
```